### ISSUE TITLE
When inferring types for the graphql schema, log when we discard a field due to conflicting types

### ISSUE THREAD

**Issue by KyleAMathews** [gatsbyjs/gatsby#3856]
Currently we just silently discard the field which causes confusion as the field just disappears. We should instead log out what happened and the fields that conflict.

**Comment by pieh** [comment]
Note to self: log it just once as we extract example fields 4 times ( 2 schema generation x (field type + field input types) )

**Comment by cricrio** [comment]
Isn't this issue the same as #2405? Because I was willing to do it but it seems really close this issue.

**Comment by pieh** [comment]
Yeah, it's same thing. Sometimes hard to keep track of things when there is single Issues tab for all kinds of things (feature request, bug reports, etc) and not everything is labeled :)

You are welcome to join conversation on linked PR ( #3905 )
